### ISSUE TITLE
Placeholders don't work in pre-release-hook

### ISSUE THREAD

**Issue by ndarilek** [crate-ci/cargo-release#469]
I have the following in my Cargo.toml:

```toml
pre-release-hook = ["git-cliff", "-o", "CHANGELOG.md", "--tag", "{{version}}"]
```

The goal is to output a CHANGELOG.md with the new version in place. Unfortunately, {{version}} is output in place of the number.

My understanding from the docs is that {{version}} should be replaced in configuration values. Either I'm misreading those, or it isn't replaced in this particular context.

Thanks.

**Comment by epage** [comment]
I've now clarified which fields are templated in the docs which doesn't include pre-release hooks.

So far, we've intended pre-release-hook to be customized via env variables.

**Comment by ndarilek** [comment]
Was going to file another issue that $NEW_VERSION doesn't work in place 
of {{version}} either. Not sure if it should.

**Comment by epage** [comment]
Are you saying ti doesn't work within the config file or the env variable isn't passed to the script?

`pre-release-hook ` is not a shell line and does not get evaluated like it (no globs, env variable replacements, etc).  The env variable is passed into the process to be read.

**Comment by ndarilek** [comment]
OK, that's fair, I can write a shell script to do this.

I *really* wish I could use `{{version}}` in that place, though. I 
pretty much just need that one command, and it's a bit easier than 
adding a shell script/adding that script to manifest distribution 
exceptions. Not a huge deal, just a papercut. Would it be overly 
difficult to support? If so, thanks for at least clarifying the 
documentation.

Thanks for the response and help.

**Comment by epage** [comment]
Been thinking about this and decided to implement it.  Its released in 0.21

**Comment by ndarilek** [comment]
Oh, cool, thanks! I'll give it a shot and report any issues I find.
